windows/src/developer/TIKE/debug/UfrmDebugStatus_Options.pas

```diff
@@ -0,0 +1,192 @@
+    lblPlatformUI: TLabel;
+    lblPlatformFormFactor: TLabel;
+    lblPlatformApplication: TLabel;
+    lblPlatformBrowser: TLabel;
+    cbPlatformOS: TComboBox;
+    cbPlatformUI: TComboBox;
+    cbPlatformFormFactor: TComboBox;
+    cbPlatformApplication: TComboBox;
+    cbPlatformBrowser: TComboBox;
+    procedure cbPlatformClick(Sender: TObject);
+    procedure FormCreate(Sender: TObject);
+  private
+    procedure SetPlatform;
+    procedure EnableControlsAndSetValues;
+  protected
+    procedure DebugKeyboardChanged; override;
+  end;
+
+implementation
+
+{$R *.dfm}
+
+// Matches https://help.keyman.com/developer/language/reference/platform
+
+const
+  PlatformUI: array[0..1] of string = (
+    'Touch',
+    'Hardware'
+  );
+  PLATFORM_UI_TOUCH = 0;
+  PLATFORM_UI_HARDWARE = 1;
+
+  PlatformOS: array[0..4] of string = (
+    'Windows',
+    'macOS',
+    'Linux',
+    'Android',
+    'iOS'
+  );
+  PLATFORM_OS_WINDOWS = 0;
+  PLATFORM_OS_MAC = 1;
+  PLATFORM_OS_LINUX = 2;
+  PLATFORM_OS_ANDROID = 3;
+  PLATFORM_OS_IOS = 4;
+
+  PlatformFormFactor: array[0..2] of string = (
+    'Desktop',
+    'Tablet',
+    'Phone'
+  );
+  PLATFORM_FF_DESKTOP = 0;
+  PLATFORM_FF_TABLET = 1;
+  PLATFORM_FF_PHONE = 2;
+
+  PlatformApplication: array[0..1] of string = (
+    'Native',
+    'Web'
+  );
+  PLATFORM_APP_NATIVE = 0;
+  PLATFORM_APP_WEB = 1;
+
+  PlatformBrowser: array[0..6] of string = (
+    '',
+    'IE',
+    'Chrome',
+    'Firefox',
+    'Safari',
+    'Opera',
+    'Edge'
+  );
+  PLATFORM_BROWSER_NA = 0;
+  PLATFORM_BROWSER_IE = 1;
+  PLATFORM_BROWSER_CHROME = 2;
+  PLATFORM_BROWSER_FIREFOX = 3;
+  PLATFORM_BROWSER_SAFARI = 4;
+  PLATFORM_BROWSER_OPERA = 5;
+  PLATFORM_BROWSER_EDGE = 6;
+
+procedure TfrmDebugStatus_Options.cbPlatformClick(Sender: TObject);
+begin
+  EnableControlsAndSetValues;
+  SetPlatform;
+  memoDebug.SetFocus;
+end;
+
+procedure TfrmDebugStatus_Options.DebugKeyboardChanged;
+begin
+  if debugkeyboard = nil then
+    Exit;
+  EnableControlsAndSetValues;
+  SetPlatform;
+end;
+
+procedure TfrmDebugStatus_Options.EnableControlsAndSetValues;
+begin
+  // Here we enable the specific allowable combinations of various
+  // platform variables. These may change over time, e.g. if we eventually
+  // support touch on Windows
+
+  case cbPlatformOS.ItemIndex of
+    PLATFORM_OS_WINDOWS, PLATFORM_OS_MAC, PLATFORM_OS_LINUX:
+    begin
+      cbPlatformUI.ItemIndex := PLATFORM_UI_HARDWARE;
+      cbPlatformUI.Enabled := False;
+      cbPlatformFormFactor.ItemIndex := PLATFORM_FF_DESKTOP;
+      cbPlatformFormFactor.Enabled := False;
+    end;
+    PLATFORM_OS_IOS:
+    begin
+      cbPlatformUI.ItemIndex := PLATFORM_UI_HARDWARE;
+      cbPlatformUI.Enabled := False;
+      if cbPlatformFormFactor.ItemIndex = PLATFORM_FF_DESKTOP then
+        cbPlatformFormFactor.ItemIndex := PLATFORM_FF_TABLET;
+      cbPlatformFormFactor.Enabled := True;
+    end;
+    PLATFORM_OS_ANDROID:
+    begin
+      cbPlatformUI.Enabled := True;
```

Comment:
Why is this `True` when PLATFORM_OS_IOS is `False`?

I also would have expected PLATFORM_OS_WINDOWS to have True too.

Reply:
Only Android devices let us switch between Hardware and Touch modes. I note that the iOS one should have been `TOUCH` not `HARDWARE`...